### PR TITLE
transfer native assets & parachain assets to chain

### PR DESCRIPTION
**Pull request by goldenfiredo**

### CODE REVIEW

File: pallets/phala/src/lib.rs [hunk 1]
```diff
@@ -16,6 +16,8 @@ use frame_support::{
 };
 use codec::Decode;
 
+//use xcm_adapter::{XcmExecutor};
+
```
Comment (h4x3rotab) [inline pallets/phala/src/lib.rs:20]: Are we waiting for the real XcmExecutor from Rococo / Polkadot? If not, we can remove this line

File: pallets/phala/src/lib.rs [hunk 2]
```diff
@@ -508,6 +516,67 @@ decl_module! {
...
+				&transfer_data.data.dest,
+				transfer_data.data.dest_network,
+				transfer_data.data.amount
+			);
+
+			// Announce the successful execution
+			IngressSequence::insert(CONTRACT_ID, sequence + 1);
+			//if let Err(_) = T::XcmExecutor::execute_xcm(MultiLocation::Null, xcm) {
+			//	Self::deposit_event(RawEvent::XcmExecutorFailed(transfer_data.data.dest.clone(), transfer_data.data.x_currency_id.clone().into(), transfer_data.data.amount, sequence + 1));
+			//}
```
Comment (h4x3rotab) [inline pallets/phala/src/lib.rs:574]: Let's add a `//TODO` here

File: pruntime/enclave/src/system.rs [hunk 3]
```diff
@@ -29,6 +29,8 @@ pub enum TransactionStatus {
     BadSecret,
     BadMachineId,
     FailedToSign,
+    BadDecimal,
+    DestroyNotAllowed,
```
Comment (h4x3rotab) [inline pruntime/enclave/src/system.rs:33]: Where do we use this?

File: pruntime/enclave/src/lib.rs [hunk 4]
```diff
@@ -1425,8 +1426,30 @@ fn parse_events(block_with_events: &BlockHeaderWithEvents, ecdh_privkey: &EcdhKe
                         };
                         handle_execution(state, &pos, who.clone(), *contract_id, payload, *num, ecdh_privkey);
                     },
-                    _ => {
+                    phala::RawEvent::TransferToTee(_, _) => {
+                        state.contract2.handle_event(evt.event.clone());
+                    },
+                    phala::RawEvent::TransferToChain(_, _, _) => {
                         state.contract2.handle_event(evt.event.clone());
+                    },
```
Comment (h4x3rotab) [inline pruntime/enclave/src/lib.rs:1434]:
```suggestion
                    phala::RawEvent::TransferToTee(_, _) | phala::RawEvent::TransferToChain(_, _, _) => {
                        state.contract2.handle_event(evt.event.clone());
                    },
```

File: pruntime/enclave/src/lib.rs [hunk 5]
```diff
@@ -1425,8 +1426,30 @@ fn parse_events(block_with_events: &BlockHeaderWithEvents, ecdh_privkey: &EcdhKe
...
+                    phala::RawEvent::TransferTokenToTee(_, _, _) => {
+                        state.contract3.handle_event(evt.event.clone());
+                    },
+                    phala::RawEvent::TransferTokenToChain(_, _, _, _) => {
+                        state.contract3.handle_event(evt.event.clone());
+                    },
+                    phala::RawEvent::TransferXTokenToChain(_, _, _, _) => {
+                        state.contract3.handle_event(evt.event.clone());
+                    },
```
Comment (h4x3rotab) [inline pruntime/enclave/src/lib.rs:1443]:
```suggestion
                    phala::RawEvent::TransferTokenToTee(_, _, _) | phala::RawEvent::TransferTokenToChain(_, _, _, _) | phala::RawEvent::TransferXTokenToChain(_, _, _, _) => {
                        state.contract3.handle_event(evt.event.clone());
                    },
```

Comment (h4x3rotab) [reply pruntime/enclave/src/lib.rs:1443]: Also take care of the indent: https://doc.rust-lang.org/1.0.0/style/style/whitespace.html

File: pruntime/enclave/src/system.rs (hunk 3 above)
Comment (goldenfiredo) [reply pruntime/enclave/src/system.rs:33]: asset contract

**Review by h4x3rotab — APPROVED** [review]

**Pull request merged**